controllers/webhook/status_updater.go

```diff
@@ -0,0 +1,133 @@
+package webhook
+
+import (
+	"context"
+	pipelinesv1 "github.com/sky-uk/kfp-operator/apis/pipelines/v1alpha5"
+	"github.com/sky-uk/kfp-operator/argo/common"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+func completionStateForRunCompletionStatus(rcs common.RunCompletionStatus) *pipelinesv1.CompletionState {
+	switch rcs {
+	case common.RunCompletionStatuses.Succeeded:
+		return &pipelinesv1.CompletionStates.Succeeded
+	case common.RunCompletionStatuses.Failed:
+		return &pipelinesv1.CompletionStates.Failed
+	default:
+		return nil
+	}
+}
+
+type StatusUpdater struct {
+	K8sClient client.Client
+}
+
+func NewStatusUpdater(ctx context.Context, scheme *runtime.Scheme) (StatusUpdater, error) {
+	logger := log.FromContext(ctx)
+	k8sConfig, err := common.K8sClientConfig()
+	if err != nil {
+		logger.Error(err, "Error reading k8s client config")
+		return StatusUpdater{}, err
+	}
+
+	k8sClient, err := client.New(k8sConfig, client.Options{
+		Scheme: scheme,
+	})
+	if err != nil {
+		logger.Error(err, "Error creating k8s client")
+		return StatusUpdater{}, err
+	}
+	logger.Info("StatusUpdater created")
+	return StatusUpdater{k8sClient}, nil
+}
+
+func (su StatusUpdater) handle(
+	ctx context.Context,
+	event common.RunCompletionEvent,
+) error {
+	if event.RunName != nil {
+		if err := su.completeRun(ctx, event); err != nil {
+			return err
+		}
+	}
+	if event.RunConfigurationName != nil {
+		if err := su.completeRunConfiguration(ctx, event); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (su *StatusUpdater) completeRun(ctx context.Context, event common.RunCompletionEvent) error {
```

Comment:
We have a mixture of pointer receiver methods and not, this is bad practice

Reply:
8873223cdbe716463093424c66e8e762392db3e5

Reply:
I guess since I assume k8client doesnt hold much state it isnt much to create a copy. Intellij seems to always default to pointer receivers. For this case I think what you have done is correct